lms/lmsweb/views.py

```diff
@@ -343,41 +363,51 @@ def upload_page():
     })
 
 
-@webapp.route(f'{routes.DOWNLOADS}/<int:solution_id>')
-@webapp.route(f'{routes.DOWNLOADS}/<int:solution_id>/<int:file_id>')
+@webapp.route(f'{routes.DOWNLOADS}/<string:endpoint>')
 @login_required
-def download(solution_id: int, file_id: Optional[int] = None):
-    solution = Solution.get_or_none(solution_id)
-    if solution is None:
+def download(endpoint: str):
+    solution = Solution.get_or_none(Solution.id == endpoint)
```

Comment:
Nope. Don't allow the share id to be downloaded, allow it only using the solution id.

The current situation might cause a collision for 10 digits long solution ids.

Download should only be allowed using the solution_id.